src/datasets/arrow_writer.py

```diff
@@ -337,38 +365,18 @@ def write_examples_on_file(self):
         if not self.current_examples:
             return
 
-        # Since current_examples contains (example, key) tuples
+        # order the columns properly
         cols = (
             [col for col in self.schema.names if col in self.current_examples[0][0]]
             + [col for col in self.current_examples[0][0].keys() if col not in self.schema.names]
             if self.schema
             else self.current_examples[0][0].keys()
         )
-
-        schema = None if self.pa_writer is None and self.update_features else self.schema
-        try_schema = self.schema if self.pa_writer is None and self.update_features else None
-        arrays = []
-        inferred_types = []
+        batch_examples = {}
         for col in cols:
-            col_type = schema.field(col).type if schema else None
-            col_try_type = try_schema.field(col).type if try_schema is not None and col in try_schema.names else None
-            typed_sequence = OptimizedTypedSequence(
-                [row[0][col] for row in self.current_examples], type=col_type, try_type=col_try_type, col=col
-            )
-            pa_array = pa.array(typed_sequence)
-            inferred_type = pa_array.type
-            first_example = pa.array(OptimizedTypedSequence(typed_sequence.data[:1], type=inferred_type))[0]
-            if pa_array[0] != first_example:  # Sanity check (check for overflow in StructArray or ListArray)
-                # This check fails with FloatArrays with nans, which is not what we want, so account for that:
-                if not isinstance(pa_array[0], pa.lib.FloatScalar):
-                    raise OverflowError(
-                        f"There was an overflow in the {type(pa_array)}. Try to reduce writer_batch_size to have batches smaller than 2GB"
-                    )
```

Comment:
I temporarily removed this part but I guess this isn't fixed yet on the arrow side ?
So I guess I'll add it back

By any chance, do you know if there's a way to test if the issue still occurs @mariosasko ? I can't find any test for that

Reply:
I was able to find the test that was used to check if an overflow was happening:
```python
import pyarrow as pa
pa.array([["x" * 1024]] * ((4 << 20) + 1))
```
From pyarrow 3 it doesn't overflow anymore, so I think we can simply remove this check now !